test/SacToken.test.js

```diff
@@ -0,0 +1,67 @@
+const SacToken = artifacts.require("SacToken");
+const UserContract = artifacts.require("UserContract");
+
+const expectRevert = require("@openzeppelin/test-helpers/src/expectRevert");
+
+contract("SacToken", (accounts) => {
+  let instance;
+  let userContract;
+  let [ownerAddress, user1Address, user2Address] = accounts;
+
+  let args = {
+    totalSacTokens: "1500000000000000000000000000",
+  };
+
+  beforeEach(async () => {
+    instance = await SacToken.new(args.totalSacTokens);
+    userContract = await UserContract.new();
+  });
+
+  context("when deploy the token contract", () => {
+    it("total suply should be equal to 1500000000000000000000000000", async () => {
+      const totalSupply = await instance.totalSupply();
+      assert.equal(totalSupply, args.totalSacTokens);
+    });
+
+    it("balance of contract owner should be equal to 1500000000000000000000000000", async () => {
+      const ownerBalance = await instance.balanceOf(ownerAddress);
+      assert.equal(ownerBalance, args.totalSacTokens);
+    });
+  });
+
+  describe("#transfer", () => {
+    context("when a user transfer 100 sac token", () => {
```

Comment:
Dois pontos:
- Podemos atualizar para: `when a user transfer 100000000000000000000 sac token`
- Aqui vale um spec pra checar se o usuário que transferiu está com 100 tokens a menos

OBS: Onde tem referência ao número 100 podemos atualizar para 100000000000000000000

Reply:
ponto 1 atualizado.
Já o ponto 2, o it da linha 51 não faz exatamente isso? Ele checa se o usuário que transferiu está com 100 tokens a menos. Eu queria adicionar também um pra checar se o address(0) está com 100 tokens a mais, mas ainda não consegui. Vou tentar fazer esse.

Reply:
`Já o ponto 2, o it da linha 51 não faz exatamente isso? ` Então, aqui estamos testando o método transfer e as ações dele.

No `it` da linha 33 vc está checando se o usuário recebeu os `100 tokens`, mas não está testando se o usuário que transferiru está com `100 tokens a menos`

Reply:
describe("#transfer", () => {
  context("when user have tokens", () => {
    context("when a user transfer 100000000000000000000 sac token", () => {
      it("it should transfer when user has tokens", async () => {
        await instance.transfer(user1Address, "100000000000000000000");
        const balanceOf = await instance.balanceOf(user1Address);
        assert.equal(balanceOf, "100000000000000000000");
      });
    });

  })

  context("when user doesn't have tokens", () => {
    it("must return erro message", async () => {
      await expectRevert(
        instance.transfer(user1Address, "100000000000000000000", { from: user2Address }),
        "Insufficient balance."
      );
    });
  });
});

Reply:
Beleza!

Reply:
Atualizado!